### PR TITLE
The version script doesn't work as is

### PR DESCRIPTION
**Pull request by pprindeville**
It's also more complicated than it needs to be.

Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>

### CODE REVIEW

File: verc.sh [hunk 1]
```diff
@@ -1,5 +1,3 @@
 #!/bin/sh
 
-printf 'const char version[] = "'
-./vers.sh
-printf '";\n'
+echo "const char version[] = \"`./vers.sh`\";"
```
Comment (JensRantil) [inline verc.sh]: I don't think the above is simple. However, I do think
```
echo -n 'const char version[] = "'
./vers.sh
echo ';'
```
is.

**Comment by pprindeville** [comment]
@JensRantil Please re-review.

**Comment by kr** [comment]
Reading the diff and comments above, it’s not entirely clear to me what the problem is.

Do you think you can briefly describe:

• steps to reproduce the problem
• the behavior you saw
• the behavior you expected

Note that the printf shell builtin is more portable than echo. As you’ve seen, echo has different usage depending on the environment. Using printf instead keeps things simpler, because there’s no need for a conditional check.

Of course it’s possible these scripts are using printf incorrectly. In that case, it’s probably easier to fix the printf usage rather than replacing it with echo.

**Comment by ysmolsky** [comment]
I second @kr opinion, it is not clear what is broken with printf. An example would be great.

**Comment by pprindeville** [comment]
> Reading the diff and comments above, it’s not entirely clear to me what the problem is.

@kr Sorry, it was a year ago that I was having this issue.  I can't remember if the problems with building were on Ubuntu 16.05, MacOS 10.13, or Fedora 28...  I'll see if I can reproduce it.

**Comment by ysmolsky** [comment]
Closing, please reopen if you can reproduce the issue.

**Pull request closed without merging**